sklearn/tests/test_impute.py

```diff
@@ -76,6 +78,24 @@ def test_imputation_shape():
         assert X_imputed.shape == (10, 2)
 
 
+@pytest.mark.parametrize("X_data, missing_value, fill_value, dtype, match",
+                         [(1, 0, "x", None, "imputing numerical"),
+                          (1., np.nan, "x", None, "imputing numerical"),
+                          ("a", "", 0, object, "imputing categorical"),
+                          (True, "nan", "x", "c", "cannot work")])
+def test_imputation_error_invalid_types(X_data, missing_value,
```

Comment:
already long name but maybe add "constant" in the name, as this is specific for that strategy?